### PR TITLE
story(TG-249): Implemented Settings -> Roles

### PR DESCRIPTION
**Pull request by yakimko**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-frontend/pull/183" target="_blank">ReviewNinja

File: src/app/settings/roles/roles.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+export class RolesComponent {
+    private subscription: Subscription;
+    private users = [];
+
+    constructor(private rolesService: RolesService) {}
+
+    ngOnInit() {
+        this.subscription = this.rolesService.retrieve().subscribe(users => {
+            this.users = users.responses;
+        });
```
Comment (sergii-trotsiuk) [inline src/app/settings/roles/roles.component.ts:17]: May be it should better hide responses in service?

File: src/app/settings/roles/roles.service.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import { ShHttpService } from '../../core/sh-http/sh-http.service';
+
+@Injectable()
+export class RolesService {
+    constructor(private http: ShHttpService) {}
+
+    retrieve() {
+        return this.http.get('/user');
+    }
```
Comment (sergii-trotsiuk) [inline src/app/settings/roles/roles.service.ts:12]: Why role service retrieve and returns users?

Comment (yakimko) [reply src/app/settings/roles/roles.service.ts:12]: discussed

File: src/app/settings/roles/roles.component.ts (hunk 1 above)
Comment (yakimko) [reply src/app/settings/roles/roles.component.ts:17]: fixed

File: src/app/settings/roles/roles.service.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        http = jasmine.createSpyObj('http', ['get']);
+
+        sut = new RolesService(http);
+
+        sut.http.get.and.returnValue(Observable.of({
+            responses: [1, 2]
+        }));
+    });
+
+    it('should be defined', () => {
```
Comment (disist) [inline src/app/settings/roles/roles.service.test.ts:17]: It's redundant

Comment (borys-ihnatyev) [reply src/app/settings/roles/roles.service.test.ts:17]: yep, the second ones can indicate that sut is defined by throwing an error

**Pull request merged**